public/resources/core/js/services/shopping-cart.service.js

```diff
@@ -46,10 +46,26 @@
         };
 
         function addItem(item) {
-
+            var headers = {};
+
+            if ('termsAccepted' in item) {
+                headers['X-Terms-Accepted'] = !!item.termsAccepted;
+                delete item.termsAccepted;
+            }
+
+            var resourceWithHeaders = $resource(URLS.SHOPPING_CART, {
+                id: '@id',
+                action: '@action'
+            },
+            {
+                save: {
+                    method: 'POST',
+                    headers: headers,
+                },
+            });
```

Comment:
This is not the place where you have to inject the headers. This method is used to save items in the persistent shopping cart (that may be removed later), You have to inject the header when the order is created and the checkout is done, when all the related items have the terms accepted. 

Indeed, I am almost sure that with this code the purchases do not work anymore, since the header is now validated in the backend and not included in the request, please try the system before commiting!